src/annotator/anchoring/pdf.ts

```diff
@@ -660,19 +650,35 @@ async function anchorShape(
     case 'rect':
       {
         const s = shapeSelector.shape;
+        let [left, top] = viewport.convertToViewportPoint(s.left, s.top);
+        let [right, bottom] = viewport.convertToViewportPoint(
+          s.right,
+          s.bottom,
+        );
+        if (right < left) {
+          [left, right] = [right, left];
+        }
+        if (bottom < top) {
+          [top, bottom] = [bottom, top];
+        }
```

Comment:
Normalize the shape here so code that uses the shape anchors (eg. to draw highlights) can assume `right > left` and `bottom > top`.